mmv1/products/datacatalog/terraform.yaml

```diff
@@ -107,12 +107,24 @@ overrides: !ruby/object:Overrides::ResourceOverrides
       fields.type.primitiveType: !ruby/object:Overrides::Terraform::PropertyOverride
         description: |
           {{description}} Exactly one of `primitive_type` or `enum_type` must be set
+        default_from_api: true
+      fields.description: !ruby/object:Overrides::Terraform::PropertyOverride
+        default_from_api: true
+      fields.displayName: !ruby/object:Overrides::Terraform::PropertyOverride
+        default_from_api: true
+      fields.isRequired: !ruby/object:Overrides::Terraform::PropertyOverride
+        default_from_api: true
+      fields.order: !ruby/object:Overrides::Terraform::PropertyOverride
+        default_from_api: true
```

Comment:
Question: why are we adding `default_from_api: true` for these fields? Will API return these values if they are not set explicitly by users?

Reply:
While discussing the fix, we discussed with Stephen that we needed to keep the change as user-friendly as possible. Since the attribute has a List datatype, altering any of the fields appears in the Terraform plan as though we are updating all the fields (like deleting + recreating all fields block), which is confusing to the end user because, on the inside, the code fix is performing an in-place change for the particular field and not all the fields block. As a workaround, we came up with the idea that when we keep it to Computed as true, it will only display the value that is currently being updated for the relevant field and other field attributes will be marked as `known after apply`, making it a little easier for users to understand what is happening behind the plan.

```
Terraform will perform the following actions:

  # google_data_catalog_tag_template.basic_tag_template will be updated in-place
  ~ resource "google_data_catalog_tag_template" "basic_tag_template" {
        id              = "projects/crest-359621/locations/us-central1/tagTemplates/my_template_1"
        name            = "projects/crest-359621/locations/us-central1/tagTemplates/my_template_1"
        # (5 unchanged attributes hidden)

      - fields {
          - display_name = "Number of rows in the data asset Test Field Update" -> null
          - field_id     = "num_rows" -> null
          - is_required  = false -> null
          - name         = "projects/crest-359621/locations/us-central1/tagTemplates/my_template_1/fields/num_rows" -> null
          - order        = 0 -> null

          - type {
              - primitive_type = "DOUBLE" -> null
            }
        }
      + fields {
          + description  = (known after apply)
          + display_name = "Number of rows in the data asset Test Field Update"
          + field_id     = "num_rows"
          + is_required  = (known after apply)
          + name         = (known after apply)
          + order        = (known after apply)

          + type {
              + primitive_type = "STRING"
            }
        }

        # (2 unchanged blocks hidden)
    }

```